src/entities/trades/swapr-v3/SwaprV3.ts

```diff
@@ -0,0 +1,209 @@
+          return null
+        })
+
+      const fee =
+        routes?.length > 0 && routes[0].pools.length > 0
+          ? new Percent(routes[0].pools[0].fee.toString(), '1000000')
+          : new Percent('0', '0')
+
+      if (quotedAmountOut) {
+        return new SwaprV3Trade({
+          maximumSlippage,
+          inputAmount: amount,
+          outputAmount: new TokenAmount(quoteCurrency, quotedAmountOut),
+          tradeType: tradeType,
+          chainId: chainId,
+          priceImpact: new Percent('0', '1000'), // todo: fix this
+          fee,
+        })
+      }
+    } else {
+      const routes: Route<Currency, Currency>[] = await getRoutes(tokenIn, tokenOut, chainId)
+
+      const fee = new Percent(routes[0].pools[0].fee.toString(), '1000000')
+
+      const quotedAmountIn = await getQuoterContract()
+        .callStatic.quoteExactOutputSingle(
+          quoteCurrency.address,
+          amount.currency.address,
+          parseUnits(amount.toSignificant(), amount.currency.decimals),
+          0
+        )
+        .catch((error) => {
+          console.error(`Error sending quoteExactOutputSingle transaction: ${error}`)
+          return null
+        })
+
+      if (quotedAmountIn) {
+        return new SwaprV3Trade({
+          maximumSlippage,
+          inputAmount: new TokenAmount(quoteCurrency, quotedAmountIn),
+          outputAmount: amount,
+          tradeType: tradeType,
+          chainId: chainId,
+          priceImpact: new Percent('0', '1000'), // todo: fix this
+          fee: fee,
+        })
+      }
+    }
+
+    return null
+  }
+
+  public minimumAmountOut(): CurrencyAmount {
+    if (this.tradeType === TradeType.EXACT_OUTPUT) {
+      return this.outputAmount
+    } else {
+      const slippageAdjustedAmountOut = new Fraction(ONE)
+        .add(this.maximumSlippage as Fraction)
+        .invert()
+        .multiply(this.outputAmount.raw).quotient
+      return this.outputAmount instanceof TokenAmount
+        ? new TokenAmount(this.outputAmount.token, slippageAdjustedAmountOut)
+        : CurrencyAmount.nativeCurrency(slippageAdjustedAmountOut, this.chainId)
+    }
+  }
+
+  public maximumAmountIn(): CurrencyAmount {
+    if (this.tradeType === TradeType.EXACT_INPUT) {
+      return this.inputAmount
+    } else {
+      const slippageAdjustedAmountIn = new Fraction(ONE)
+        .add(this.maximumSlippage as Fraction)
+        .multiply(this.inputAmount.raw).quotient
+      return this.inputAmount instanceof TokenAmount
+        ? new TokenAmount(this.inputAmount.token, slippageAdjustedAmountIn)
+        : CurrencyAmount.nativeCurrency(slippageAdjustedAmountIn, this.chainId)
+    }
+  }
+
+  public async swapTransaction(options: TradeOptions): Promise<UnsignedTransaction> {
+    const to: string = validateAndParseAddress(options.recipient)
+    const amountIn: string = toHex(this.maximumAmountIn())
+    const amountOut: string = toHex(this.minimumAmountOut())
```

Comment:
EthersJS has this conversion. You need to use `ethers.utils.hexlify`

Reply:
Seems not be working.